src/components/content/credential/UpdateCredential.tsx

```diff
@@ -0,0 +1,214 @@
+    const handleCspSelect = (cspName: CredentialVariables.csp) => {
+        setCurrentCsp(cspName);
+    };
+
+    const handleCredentialTypeSelect = (type: CredentialVariables.type) => {
+        setCurrentType(type);
+    };
+
+    const setFormFields = () => {
+        form.setFieldsValue({ name: createCredential.name });
+        form.setFieldsValue({ csp: createCredential.csp });
+        form.setFieldsValue({ xpanseUser: createCredential.xpanseUser });
+        form.setFieldsValue({ description: createCredential.description });
+        form.setFieldsValue({ type: createCredential.type });
+        form.setFieldsValue({ variables: createCredential.variables });
+        form.setFieldsValue({ timeToLive: createCredential.timeToLive });
+    };
+
+    const submit = (credentialVariables: CredentialVariables) => {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+        const createCredential: CreateCredential = JSON.parse(JSON.stringify(credentialVariables));
+        createCredential.timeToLive = 3600;
+        const userName: string | null = localStorage.getItem(usernameKey);
+        if (userName === null) {
+            return;
+        }
+        createCredential.xpanseUser = userName;
+        void CredentialsManagementService.updateCredential(createCredential)
+            .then(() => {
+                getCredentials();
+                getTipInfo('success', 'Update Credential Success!');
+                onUpdateCancel();
+            })
+            .catch(() => {
+                getTipInfo('error', 'Update Credential Failed!');
+            });
+    };
+
+    const tailLayout = {
+        wrapperCol: { offset: 8, span: 16, margin: 100 },
+    };
+
+    const setVariablesValue = (index: number, e: ChangeEvent<HTMLInputElement>) => {
+        credentialVariableList[index].value = e.target.value;
+        form.setFieldsValue({ variables: credentialVariableList });
+    };
+
+    useEffect(() => {
+        setFormFields();
+        if (currentCsp === undefined) {
+            return;
+        }
+        void CredentialsManagementService.getCredentialTypesByCsp(currentCsp).then((resp) => {
+            if (resp.length > 0) {
+                setCredentialTypeList(resp as CredentialVariables.type[]);
+            }
+        });
+        if (currentType === undefined) {
+            return;
+        }
+        void CredentialsManagementService.getCredentialCapabilitiesByCsp(currentCsp, currentType).then((resp) => {
+            if (resp.length > 0) {
+                const credentialVariables: CredentialVariables = resp[0] as CredentialVariables;
+                setCredentialVariableList(credentialVariables.variables);
+                form.setFieldsValue({ variables: credentialVariables.variables });
+            }
+        });
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [currentCsp, currentType]);
+
+    const columns: ColumnsType<CredentialVariable> = [
```

Comment:
mandatory fields must be marked as mandatory.

Reply:
@swaroopar  HI swaroopar,
I didn't understand it, when we click to modify, the original data should be echoed, there shouldn't be no value here

Reply:
Hi.. you are right.  Pls ignore this comment.